server/sonar-web/src/main/js/apps/projectQualityGate/App.tsx

```diff
@@ -82,7 +82,7 @@ export default class App extends React.PureComponent<Props> {
     );
   }
 
-  handleChangeGate = (oldId: string | undefined, newId: string | undefined) => {
+  handleChangeGate = (oldId: number | undefined, newId: number | undefined) => {
```

Comment:
I think you can use `(oldId?: number, newId?: number)`

Reply:
this is tricky: `oldId?: number` means the parameter can be omitted, `oldId: number | undefined` means the parameter must be provided, but can have value `undefined`